src/tests/test_rng.cpp

```diff
@@ -135,11 +191,209 @@ class HMAC_DRBG_Reseed_Tests : public Test
          // reseed must happend here
          rng.randomize(out.data(), out.size());
          result.test_ne("out after reseed", out, output_without_reseed);
-         return {result};
+         return result;
+         }
+
+      Test::Result test_broken_entropy_input()
+         {
+         Test::Result result("HMAC_DRBG Broken Entropy Input");
+
+         auto mac = Botan::MessageAuthenticationCode::create("HMAC(SHA-256)");
+         if(!mac)
+            {
+            result.note_missing("HMAC(SHA-256)");
+            return result;
+            }
+
+         // make sure no output is generated when the entropy input source is broken
+
+         // underlying_rng throws exception
+         Botan::Null_RNG broken_entropy_input_rng;
+         Botan::HMAC_DRBG rng_with_broken_rng(std::move(mac), broken_entropy_input_rng, 16384);
+         result.test_throws("broken underlying rng", [&rng_with_broken_rng] () { rng_with_broken_rng.random_vec(16); });
+
+         // entropy_sources throw exception
+         std::unique_ptr<Broken_Entropy_Source> broken_entropy_source_1(new Broken_Entropy_Source());
+         std::unique_ptr<Broken_Entropy_Source> broken_entropy_source_2(new Broken_Entropy_Source());
+
+         Botan::Entropy_Sources broken_entropy_sources;
+         broken_entropy_sources.add_source(std::move(broken_entropy_source_1));
+         broken_entropy_sources.add_source(std::move(broken_entropy_source_2));
+
+         mac = Botan::MessageAuthenticationCode::create("HMAC(SHA-256)");
+         Botan::HMAC_DRBG rng_with_broken_es(std::move(mac), broken_entropy_sources, 16384);
+         result.test_throws("broken entropy sources", [&rng_with_broken_es] () { rng_with_broken_es.random_vec(16); });
+
+         // entropy source returns insufficient entropy
+         Botan::Entropy_Sources insufficient_entropy_sources;
+         std::unique_ptr<Insufficient_Entropy_Source> insufficient_entropy_source(new Insufficient_Entropy_Source());
+         insufficient_entropy_sources.add_source(std::move(insufficient_entropy_source));
+
+         mac = Botan::MessageAuthenticationCode::create("HMAC(SHA-256)");
+         Botan::HMAC_DRBG rng_with_insufficient_es(std::move(mac), insufficient_entropy_sources, 16384);
+         result.test_throws("insufficient entropy source", [&rng_with_insufficient_es] () { rng_with_insufficient_es.random_vec(16); });
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=cpp%3AOneStatementPerLine)